libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -684,6 +688,321 @@ impl Greenlight {
+                index: Some(ListsendpaysIndex::Created.into()),
+                start: Some(sync_state.send_pays_index.created),
+                ..Default::default()
+            })
+            .await?
+            .into_inner();
+        if let Some(last) = created_send_pays.payments.last() {
+            new_sync_state.send_pays_index.created = last.created_index()
+        }
+        let updated_send_pays = node_client
+            .list_send_pays(cln::ListsendpaysRequest {
+                index: Some(ListsendpaysIndex::Updated.into()),
+                start: Some(sync_state.send_pays_index.updated),
+                ..Default::default()
+            })
+            .await?
+            .into_inner();
+        if let Some(last) = updated_send_pays.payments.last() {
+            new_sync_state.send_pays_index.updated = last.created_index()
+        }
+        let mut hashes: Vec<_> = created_send_pays
+            .payments
+            .iter()
+            .chain(updated_send_pays.payments.iter())
+            .map(|p| p.payment_hash.clone())
+            .collect();
+        hashes.dedup();
+        let hash_groups: HashSet<_> = created_send_pays
+            .payments
+            .iter()
+            .chain(updated_send_pays.payments.iter())
+            .map(|p| {
+                let mut key = hex::encode(&p.payment_hash);
+                key.push('|');
+                key.push_str(&p.groupid.to_string());
+                key
+            })
+            .collect();
+        trace!(
+            "list sendpays: created: {:?}, updated: {:?}",
+            created_send_pays,
+            updated_send_pays
+        );
+        self.persister.insert_send_pays(
+            &created_send_pays
+                .payments
+                .into_iter()
+                .map(TryInto::try_into)
+                .collect::<Result<Vec<_>, _>>()?,
+        )?;
+        self.persister.insert_send_pays(
+            &updated_send_pays
+                .payments
+                .into_iter()
+                .map(TryInto::try_into)
+                .collect::<Result<Vec<_>, _>>()?,
+        )?;
+
+        // Now all new send_pays are persisted. Retrieve the send_pays for the
+        // payment hashes, to ensure any send_pays belonging to the same payment
+        // that were not fetched in this round are also included.
+        let send_pays = self.persister.list_send_pays(&hashes)?;
+
+        // Now that all send_pays belonging to all payments are here, aggregate
+        // the send_pays into payments. This is a copy of what core lightning's
+        // listpays function does under the hood.
+        let mut outbound_payments: HashMap<String, SendPayAgg> = HashMap::new();
+        for send_pay in send_pays {
+            let mut key = hex::encode(&send_pay.payment_hash);
+            key.push('|');
+            key.push_str(&send_pay.groupid.to_string());
+            if !hash_groups.contains(&key) {
```

Comment:
Do you think it makes sense to query from the db directly those that we want? (based on the payment hash and group id).